app/src/lib/app-state.ts

```diff
@@ -771,11 +771,31 @@ export interface IChangesState {
    */
   readonly currentRepoRulesInfo: RepoRulesInfo
 
-  /** The text entered into the compare branch filter text box */
+  /** The file list filter state containing all filter options */
+  readonly fileListFilter: IFileListFilterState
+}
+
+/**
+ * State interface for file list filtering options
+ */
+export interface IFileListFilterState {
+  /** The text entered into the filter text box */
   readonly filterText: string
 
   /** The state of the changes list filter of included changes. */
   readonly includedChangesInCommitFilter: boolean
+
+  /** Whether to filter and show only new files */
+  readonly filterNewFiles: boolean
+
+  /** Whether to filter and show only modified files */
+  readonly filterModifiedFiles: boolean
+
+  /** Whether to filter and show only deleted files */
+  readonly filterDeletedFiles: boolean
+
+  /** Whether to filter and show only excluded from commit files */
+  readonly filterExcludedFiles: boolean
```

Comment:
Nit: "filterExcludedFiles" is the opposite of "includedChangesInCommitFilter" but they feel very different. Be nice to use consistent filter names across all filters. Additionally, "Exclude" feels to generic? 

Another Nit: We are grouping these in a `IFileListFilterState`, I do not think they need the word "filter" in them. Thus, we can be more concise with names like "isIncludeInCommit", "isExcludedFromCommit", "isNewFile", "isModifiedFile", "isDeletedFile".